### ISSUE TITLE
Support 'username' as an optional argument to FilesystemResolver

### ISSUE THREAD

**Issue by selitvin** [uber/petastorm#317]
Including forwarding of username argument from calling sites of FilesystemResolver where it makes sense.

**Comment by jsgoller1** [comment]
@selitvin - I am assuming that this is to enhance support for HDFS? S3 credentials should be set from the environment / `~/.aws/credentials`, and I can't think of a case where the local filesystem would need a `username`.

**Comment by selitvin** [comment]
Yep. This is for hdfs case, where the username can be specified, depends on the hdfs cluster configuration. This is not the most elegant solution since the argument is not applicable in some `FilesystemResolver` usage scenarios. Maybe we should issue a warning, if hdfs_driver or username is specified for non hdfs filesystems.

**Comment by wangqiaoshi** [comment]
I think it's a very useful feature.  If I need to proxy user to access hdfs.

**Comment by Ivan-Dimitrov** [comment]
Reference PR #384
